openslides_backend/action/actions/motion/json_upload.py

```diff
@@ -0,0 +1,712 @@
+            entry["category_name"] = {"value": "", "info": ImportState.WARNING}
+            messages.append("Category could not be found")
+
+        if number := entry.get("number"):
+            check_result = self.number_lookup.check_duplicate(number)
+            id_ = cast(int, self.number_lookup.get_field_by_name(number, "id"))
+            if check_result == ResultType.FOUND_ID and id_ != 0:
+                self.row_state = ImportState.DONE
+                set_entry_id = True
+                entry["number"] = {
+                    "value": number,
+                    "info": ImportState.DONE,
+                    "id": id_,
+                }
+            elif check_result == ResultType.NOT_FOUND or id_ == 0:
+                self.row_state = ImportState.NEW
+                entry["number"] = {
+                    "value": number,
+                    "info": ImportState.DONE,
+                }
+            elif check_result == ResultType.FOUND_MORE_IDS:
+                self.row_state = ImportState.ERROR
+                entry["number"] = {
+                    "value": number,
+                    "info": ImportState.ERROR,
+                }
+                messages.append("Error: Found multiple motions with the same number")
+        else:
+            category_id: Optional[int] = None
+            if entry.get("category_name"):
+                category_id = entry["category_name"].get("id")
+            self.row_state = ImportState.NEW
+            value: Dict[str, Any] = {}
+            self.set_number(
+                value,
+                meeting_id,
+                self._get_first_workflow_state_id(meeting_id),
+                None,
+                category_id,
+                other_forbidden_numbers=self._previous_numbers,
+            )
+            if number := value.get("number"):
+                entry["number"] = {"value": number, "info": ImportState.GENERATED}
+                self._previous_numbers.append(number)
+
+        has_submitter_error: bool = False
+        for fieldname in ["submitter", "supporter"]:
+            if users := entry.get(f"{fieldname}s_username"):
+                verbose = entry.get(f"{fieldname}s_verbose", [])
+                verbose_user_mismatch = len(verbose) > len(users)
+                username_set: Set[str] = set([])
+                entry_list: list[Dict[str, Any]] = []
+                message_map: Dict[str, Tuple[str, List[str]]] = {}
+                for user in users:
+                    if verbose_user_mismatch:
+                        entry_list.append({"value": user, "info": ImportState.ERROR})
+                    elif user in username_set:
+                        entry_list.append({"value": user, "info": ImportState.WARNING})
+                        err_message, err_users = message_map.get(
+                            "duplicate",
+                            (
+                                f"At least one {fieldname} has been referenced multiple times: ",
```

Comment:
It's better to gather the `username` in a list,  together with the `fieldname` and create the message one time at the end of the `fieldnames` loop